src/dataloader.py

```diff
@@ -0,0 +1,90 @@
+import itertools
+import os
+from dataclasses import dataclass, field
+from typing import Dict, Literal, Optional
+
+import torch
+from datasets import IterableDatasetDict, concatenate_datasets, load_dataset
+from torch.utils.data import DataLoader
+from transformers import DataCollatorForLanguageModeling, PreTrainedTokenizerBase
+
+
+@dataclass
+class BooksCorpusAndWiki:
```

Comment:
Should add `BaseDataModule` class in future, to support many datasets with a common interface. Will keep as-is for now.

Reply:
sounds good. no need to write abstractions we don't need for now.